components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2762,30 +2762,51 @@ private boolean isFilteringNotSupported(String filterOperation) {
     }
 
     /**
-     * get the search value after appending the delimiters
-     * @param filterOperation operator value
-     * @param attributeValue search value
-     * @param delimiter delimiter for based on search type
-     * @return search attribute
+     * Get the search value after appending the delimiters according to the attribute name to be filtered.
+     *
+     * @param attributeName   Filter attribute name
+     * @param filterOperation Operator value
+     * @param attributeValue  Search value
+     * @param delimiter       Filter delimiter based on search type
+     * @return Search attribute
      */
-    private String getSearchAttribute(String filterOperation, String attributeValue, String delimiter) {
+    private String getSearchAttribute(String attributeName, String filterOperation, String attributeValue,
+            String delimiter) {
 
         String searchAttribute = null;
         if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.CO)) {
-            String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR);
-            if(attributeItems.length == 2) {
-                searchAttribute = attributeItems[0] + CarbonConstants.DOMAIN_SEPARATOR +
-                        delimiter + attributeItems[1] + delimiter;
+
+            // For attributes which support domain embedding, create search value by appending the delimiter after
+            // the domain separator.
+            if (isDomainEmbeddingSupportedAttribute(attributeName)) {
+
+                // Check whether domain is embedded in the attribute value.
+                String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR);
```

Comment:
Does `isDomainEmbeddingSupportedAttribute` ensure that we only have one domain separator in the attribute value? Anyway, shall we set a split limit to `2` to make the code more readable and less error-prone?

Reply:
No it only checks whether the filter attribute name have the right to send the domain embedded with the filter attribute value. The reason for not splitting in to `2` is, we do not support `"/"` in in some attributes. Therefore, if there are more than 1 domain separator, we have considered it as not valid argument.

Reply:
`else {
                    searchAttribute = delimiter + attributeValue + delimiter;
                }`

If there are multiple `/` values, you are still considering as a valid argument here, right? Or is it handled some other place?

Reply:
Yes it's a valid argument. But if there are multiple `/` values we cannot say that 1st split is definitely a domain.

Reply:
Fixed : a9a485aaa5f32c15799ae2ceac544923ca6f2c49